modules/deployment-scripts/create-kv-certificate/test/main.test.bicep

```diff
@@ -0,0 +1,47 @@
+param location string = resourceGroup().location
+param akvName string =  'akvtest${uniqueString(newGuid())}'
+
+@description('A short retention is good for test envs')
+var shortRetention = 'PT1H'
+
+//Prerequisites
+resource akv 'Microsoft.KeyVault/vaults@2021-11-01-preview' = {
+  name: akvName
+  location: location
+  properties: {
+    sku: {
+      family: 'A'
+      name: 'standard'
+    }
+    tenantId: subscription().tenantId
+    enableSoftDelete: false
```

Comment:
Ahh this explains why the CI failed to purge the key vault. Because soft delete is disabled, the CI will never find the key vault in deleted state. To fix the CI failure we can change value to true (or just remove the property as the default is true). Key vault is also going to deprecate the ability to opt out of soft-delete soon: https://docs.microsoft.com/en-us/azure/key-vault/general/soft-delete-change.

I will update the CI to skip purging for soft-delete disabled key vaults.

Reply:
Don't we deploy each pipeline run to a unique RG?  Then it's just a matter of using uniqueString() to name resources with global names and we can let the platform deal with the purge (and authors don't have to worry about how to set this for every given resource).

Reply:
Yeah, but we may need a linter rule to enforce that. I think we should still prevent the CI from failing in case the name of the key vault is hard coded in the test file. If the key vault is not purged, the subsequent deployment will fail. This was suggested by the CARML team btw.

Reply:
> Ahh this explains why the CI failed to purge the key vault. Because soft delete is disabled, the CI will never find the key vault in deleted state. To fix the CI failure we can change value to true (or just remove the property as the default is true). Key vault is also going to deprecate the ability to opt out of soft-delete soon: https://docs.microsoft.com/en-us/azure/key-vault/general/soft-delete-change.

For the local development loop, avoiding soft delete is a must.
Since the team's going to deprecate that ability, i'll base that value off the existance of a tag to get the best experience in CI and locally.

```bicep
enableSoftDelete: ! contains(resourceGroup().tags,'kvSoftDelete')
```

Reply:
@shenglol - Another CI issue i'm afraid. Works locally, but not in CI.... The classic, "It works on my machine 👅 "

> Unable to process template language expressions for resource '/subscriptions/3ce675b8-fcea-4732-a965-393bbe02d7ad/resourceGroups/84-e23e630-20220422123703-8fe43199e63746e6bcdf864009574c0d/providers/Microsoft.KeyVault/vaults/akvtestjj2zsmhppdpww' at line '24' and column '9'. 'The language expression property 'tags' doesn't exist, available properties are 'id, name, type, location, properties'.

Reply:
@shenglol  - i've reverted the change that was causing the issue, but this might still be a good one to look at.

Reply:
According to https://github.com/MicrosoftDocs/azure-docs/issues/59033#issuecomment-662274297 the error will be returned if we try to call `resourceGroup().tags` on a resource group that was created without specifying the tags property (seems to be a bug in the resource group type🤦‍♂️). I'll change the CI to set the tags property for each test resource group.